cli/command_snapshot_create.go

```diff
@@ -329,6 +329,16 @@ func (c *commandSnapshotCreate) snapshotSingleSource(ctx context.Context, rep re
 		manifest.EndTime = endTimeOverride
 	}
 
+	ignoreIdenticalSnapshot := policyTree.EffectivePolicy().RetentionPolicy.IgnoreIdenticalSnapshots.OrDefault(false)
+	if ignoreIdenticalSnapshot {
+		for _, prev := range previous {
+			if prev.RootObjectID().String() == manifest.RootObjectID().String() {
+				log(ctx).Errorf("\n Not saving snapshot because no files have been changed")
+				return errors.Wrap(err, "Not saving snapshot because no files have been changed")
```

Comment:
unclear if you want to return an error. I would simply log.Infof() and return success.

Reply:
Beyond the logging issue you mention, what do you think about changing this code so that the root ID of the current snapshot is only compared to the root ID of the most recent snapshot rather than the root ID of all previous snapshots? Not only would this increase performance, but logically it makes sense.